src/exo/LoopIR_pprint.py

```diff
@@ -677,7 +677,8 @@ def _print_cursor_stmt(
 
     else:
         lines = _print_stmt(stmt, env, indent)
-    if cur == target:
+
+    if isinstance(target, Node) and isinstance(cur, Node) and cur._node == target._node:
```

Comment:
Why was this changed? Is there a test case that demonstrates a need for this?

Reply:
cur and target were always not equal even though they were pointing to the same node. So "# <-- NODE" was never added to the output code.

Reply:
@alexreinking Sorry, my bad. I thought this line was failing the test I wanted to add, but it wasn't related. I think this was just some leftover from debugging. Thanks for catching this!